### PR TITLE
[Bugfix] Build the forced-aligner prompt without a chat template (word timestamps one bin late)

### PR DESCRIPTION
**Pull request by twu3202**
## Purpose

`vllm_omni/utils/qwen3_force_align_processor.py::build_prompt` wraps the Qwen3-ForcedAligner input in `<|im_start|>user\n … <|im_end|>\n<|im_start|>assistant\n`. The official `qwen_asr` aligner (`Qwen3ForceAlignProcessor.encode_timestamp`) feeds `<|audio_start|><|audio_pad|><|audio_end|>` + `word<timestamp><timestamp>…` straight to the tokenizer, with no chat template (the checkpoint's own `chat_template.json` would even drop the user text).

The three tokens placed before the audio move the predicted markers late. Official transformers implementation (`qwen-asr` 0.0.6, `Qwen/Qwen3-ForcedAligner-0.6B`, fp32), same audio, text and weights, 20 FLEURS test clips (10 Mandarin + 10 English, 1014 markers), reference = official prompt:

| prompt | markers later / earlier / unchanged | mean shift |
|---|---|---|
| current `build_prompt` (prefix + suffix) | 925 / 12 / 77 | +92 ms |
| prefix `<\|im_start\|>user\n` only | 925 / 12 / 77 | +92 ms |
| suffix `<\|im_end\|>\n<\|im_start\|>assistant\n` only | 0 / 0 / 1014 | 0 |

662 markers move by exactly one 80 ms bin and 250 by two; every clip shows it (per-clip mean +52 … +137 ms, 95 % CI of the clip mean ±10 ms, sign test p ≈ 1e-255). The median word in these clips lasts 160 ms, so the bias is visible to any consumer of `word_timestamps`.

Fix: `build_prompt` returns exactly what `encode_timestamp` returns (empty word list → two markers, as upstream). vLLM's Qwen3-ASR multimodal processor only replaces `<|audio_pad|>`, so the bare prompt is valid input for the pooling stage; `code2wav2aligner` and the decoder are unchanged.

## Test Plan

**vLLM Version:** 0.28.0

**vLLM-Omni Commit:** cf36f5e4 (unit tests), 3204a0b2 (end to end); branch rebased on 087ac46

- `pytest tests/utils/test_qwen3_force_align_processor.py tests/utils/test_forced_aligner.py tests/model_executor/stage_input_processors/test_forced_aligner_stage.py` (CPU). The existing `build_prompt` test now pins the official format; a new test compares against `qwen_asr` when it is installed.
- End to end on an RTX 6000 Ada: `vllm serve Qwen/Qwen3-TTS-12Hz-1.7B-CustomVoice --omni --deploy-config <qwen3_tts.yaml with async_chunk: false> --forced-aligner Qwen/Qwen3-ForcedAligner-0.6B`, 10 `/v1/audio/speech` requests with `word_timestamps: true` and a fixed seed (5 English / Vivian, 5 Mandarin / Serena), with and without this diff; the official aligner run on each returned wav + text is the reference. (`async_chunk: false` because with the shipped yaml the aligner stage stalls "waiting for a chunk" on both branches — separate issue, unrelated to this change.)

## Test Result

- `main`: 24 passed. This branch: 24 passed, 1 skipped (no `qwen_asr` in that env; with it installed the new test passes). `ruff check` / `ruff format --check` clean. `build_prompt` equals the `encode_timestamp` output for English, Chinese, Japanese, Korean and empty text.
- End to end: the wav files are byte-identical between the two servers, so only the timestamps differ. Server `X-Word-Timestamps` vs the official aligner on the same audio, 392 markers:

  | server | later / earlier / equal | mean signed difference |
  |---|---|---|
  | `main` | 336 / 8 / 48 | +87.8 ms |
  | `main` + this PR | 20 / 35 / 337 | −2.9 ms |

  "quick brown fox", `(start_ms, end_ms)`: official `(160,560) (560,960) (960,1440)`; `main` `(160,720) (720,1120) (1120,1600)`; with this PR identical to official. The remaining single-bin differences after the fix come from the decoder's monotonic/duration clamp and bf16 (server) vs fp32 (reference).

<details>
<summary>A/B script (official transformers aligner; variants bare / wrapped / prefix_only / suffix_only)</summary>

```python
import json, statistics as st, sys, torch
from qwen_asr.inference.qwen3_forced_aligner import Qwen3ForcedAligner
from qwen_asr.inference.utils import normalize_audios

al = Qwen3ForcedAligner.from_pretrained("Qwen/Qwen3-ForcedAligner-0.6B", dtype=torch.float32, device_map="cpu")
PRE, POST = "<|im_start|>user\n", "<|im_end|>\n<|im_start|>assistant\n"
VARIANTS = {"bare": (False, False), "wrapped": (True, True), "prefix_only": (True, False), "suffix_only": (False, True)}

@torch.inference_mode()
def run(audio, text, lang, pre, post):
    word_list, input_text = al.aligner_processor.encode_timestamp(text, lang)
    input_text = (PRE if pre else "") + input_text + (POST if post else "")
    inputs = al.processor(text=[input_text], audio=[audio], return_tensors="pt", padding=True).to(al.model.device).to(al.model.dtype)
    out_ids = al.model.thinker(**inputs).logits.argmax(dim=-1)
    masked = out_ids[0][inputs["input_ids"][0] == al.timestamp_token_id]
    parsed = al.aligner_processor.parse_timestamp(word_list, (masked * al.timestamp_segment_time).cpu().numpy())
    return [(w["start_time"], w["end_time"]) for w in parsed]

for m in json.load(open("clips/meta.json")):  # [{"path", "lang", "text"}, ...] FLEURS test clips, 3-20 s
    audio = normalize_audios([m["path"]])[0]
    ts = {k: run(audio, m["text"], m["lang"], *v) for k, v in VARIANTS.items()}
    for k in ("wrapped", "prefix_only", "suffix_only"):
        d = [b - a for (a1, a2), (b1, b2) in zip(ts["bare"], ts[k]) for a, b in ((a1, b1), (a2, b2))]
        print(m["path"], k, "changed", sum(1 for x in d if x), "/", len(d), "mean %+.0f ms" % st.mean(d))
```

</details>

### CODE REVIEW

**Comment by twu3202** [comment]
### Self-review

- Scope: `build_prompt` in `vllm_omni/utils/qwen3_force_align_processor.py` now returns the same string as qwen_asr's `Qwen3ForceAlignProcessor.encode_timestamp` (audio placeholder + `word<timestamp><timestamp>…`, two markers for an empty list), with no chat-template wrapper. The existing `build_prompt` test is updated, a second test compares against `qwen_asr` when it is installed, and one comment in `deploy/qwen3_tts_forced_aligner.yaml` no longer mentions a chat template. `code2wav2aligner`, `_decode_timestamps` and the stage injection are untouched.
- What I checked: the official aligner never applies a chat template (its processor tokenizes the bare string; the checkpoint's `chat_template.json` would drop the user text). In the transformers A/B the prefix alone reproduces the +80 ms shift and the suffix alone changes nothing; end to end in vLLM-Omni the fix brings the served timestamps onto the official aligner's output (tables in the description), which also shows that the bare prompt is accepted by vLLM's Qwen3-ASR multimodal processor and the pooling stage.
- Tests: the three aligner test files plus `tests/config/test_forced_aligner_injection.py` pass on this branch rebased on 087ac46 (27 passed, 1 skipped without `qwen_asr`); ruff check/format clean.

**Review by NickCao — APPROVED** [review]
Cross validated against upstream implementation.
